pkg/topology/kademlia/kademlia.go

```diff
@@ -1340,6 +1345,17 @@ func (k *Kad) EachPeerRev(f topology.EachPeerFunc, filter topology.Filter) error
 		return f(addr, po)
 	})
 }
+func (k *Kad) PeersCount(filter topology.Filter) int {
+	count := 0
+	_ = k.connectedPeers.EachBinRev(func(addr swarm.Address, _ uint8) (bool, bool, error) {
```

Comment:
I think there is a function called `Length` on connectedPeers which gives complete length of pslice.